py/desisurvey/nextfield.py

```diff
@@ -76,4 +84,179 @@ def get_next_field(dateobs, skylevel, seeing, transparency, previoustiles,
+    
+    #Calculate the sideral time in Greenwich for the last Julian midnight
+    GMST_0 = 100.4606184+36000.77005361*T+0.00038793*T**2-2.6E-08*T**3
+    
+    #Add correction for the number of hours that have past since midnight
+    GMST = GMST_0+0.25068447733746215*(dateobs-JD_0)*24*60
+    
+    #Calculate the equation of equinoxes
+    L = 280.47+0.98565*D
+    Omega = 125.04-0.052954*D
+    delPsi = -0.000319*math.sin(Omega*math.pi/180)-0.00024*math.sin(2*L*math.pi/180)
+    E = 23.4393-0.0000004*D
+    
+    eqeq = delPsi*math.cos(E*math.pi/180)
+    
+    #Correct with the equation of equinoxes to get the current apparent sidereal time
+    GAST = GMST+eqeq
+    
+    #Add the longitude of the observatory to get the local sidereal time
+    LAST = GAST-111.5984796
+        
+    #Shift the local sidereal time into the range of 0 to 360 degrees
+    if LAST >= 360:
+        n = math.floor(LAST/360)
+        LAST = LAST-360*n
+        
+    #hour = math.floor(LAST/15)
+    #minute = math.floor((LAST/15-hour)*60)
+    #second = ((LAST/15-hour)*60-minute)*60
+    
+    #print( str(hour) + "h " + str(minute) + "m " + str(second) +"s")
+
+    #print("dateobs = " + str(dateobs))
+    #print("JD_0 = " + str(JD_0))
+    #print("D_0 = " + str(D_0))
+    #print("D = " + str(D))
+    #print("T = " + str(T))
+    #print("H = " + str(H))
+    #print("GMST = " + str(GMST))
+    #print("LAST = " + str(LAST))
+    
+    #Use PyAstronomy to calculate the position of the Sun.
+    pos_sun = pyasl.sunpos(dateobs,full_output=True)
+    
+    #print("RA_sun = " + str(float(pos_sun[1])) + " Dec_sun = " + str(float(pos_sun[2])))
+    
+    #Check to see if the Sun is up.
+    RA_sun = float(pos_sun[1])
+    Dec_sun = float(pos_sun[2])
+    HA_sun = LAST - RA_sun
+    if (HA_sun < 0):
+        HA_sun = HA_sun+360
+    if (HA_sun > 360):
+        HA_sun = HA_sun-360
+    #print("HA_sun = " + str(HA_sun))
+    ALT_sun = (math.asin(math.sin(Dec_sun*math.pi/180)*math.sin(31.9614929*math.pi/180)+math.cos(Dec_sun*math.pi/180)*math.cos(31.9614929*math.pi/180)*math.cos(HA_sun*math.pi/180)))*(180/math.pi)
+    #print("ALT_sun = " + str(ALT_sun))
+    
+    #Print warning if the Sun is up. We may decide this should do more than just warn
+    if (ALT_sun >= 0):
+        print("WARNING: The sun is currently up!")
+        
+    #Find the position of the Moon using PyAstronomy
+    pos_moon = pyasl.moonpos(dateobs)
+    
+    RA_moon = float(pos_moon[0])
+    Dec_moon = float(pos_moon[1])
+    
+    print("RA_moon = " + str(RA_moon) + " Dec_moon = " + str(Dec_moon))
+    
+    #Find the phase of the moon using PyAstronomy
+    moonPhase = pyasl.moonphase(dateobs)
+    phase = float(moonPhase)
+    
+    print(phase)
+    
+    #Opens a file containing information about all of the DESI tiles. Not sure if
+    #this is the best way to get that information?
+    tiles_file = open("desi-tiles-full.par.txt","r")
```

Comment:
Reading the tiles file should use desimodel.io.load_tiles(), which returns a numpy array with columns TILEID, RA, DEC, PASS, IN_DESI, EBV_MED, AIRMASS, EXPOSEFAC, STAR_DENSITY.